atlasdb-api/src/main/java/com/palantir/atlasdb/config/DbTimestampCreationSetting.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.config;
+
+import java.util.Optional;
+
+import org.derive4j.Data;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.TimestampSeries;
+
+@Data
+public abstract class DbTimestampCreationSetting {
+    public interface Cases<R> {
+        R multipleSeries(Optional<TableReference> tableReference, TimestampSeries series);
+        R singleSeries(Optional<TableReference> tableReference);
+    }
+
+    public abstract <R> R match(Cases<R> cases);
+
+    @Override
```

Comment:
Note: these indicate that the methods should be generated

Reply:
interesting - I guess you wouldn't want them if you aren't testing or putting them in sets, though it does feel a bit awkward that these aren't by default